proto/public/api.proto

```diff
@@ -124,8 +124,68 @@ message ApiError {
   string error = 1;
 }
 
+message ResourceSpec {
+  string namespace = 1;
+  string resource = 2;
+}
+
+message ResourceSelection {
+  ResourceSpec spec = 1;
+  string label_selector = 2;
+}
+
+message ResourceError {
+  ResourceSpec spec = 1;
+  string error = 2;
+}
+
+message StatSummaryRequest {
+  ResourceSelection resource = 1;
+  TimeWindow time_window = 2;
+
+  oneof outbound {
+    Empty none = 3;
+    ResourceSpec out_to_resource   = 4;
+    ResourceSpec out_from_resource = 5;
+  }
+}
+
+message StatSummaryResponse {
+  oneof response {
+    Ok ok = 1;
+    ResourceError error = 2;
+  }
+
+  message Ok {
+    repeated StatTable stat_tables = 1;
+  }
+}
+
+message StatTable {
+  oneof table {
+    PodGroup pod_group = 1;
+  }
+
+  message PodGroup {
+    repeated Row rows = 1;
+
+    message Row {
+      ResourceSpec spec = 1;
+
+      uint64 meshed_pod_count = 2;
+      uint64 total_pod_count = 3;
+
+      uint64 success_count = 4;
+      uint64 failure_count = 5;
+      uint64 latency_p50 = 6;
+      uint64 latency_p90 = 7;
```

Comment:
nit, but we should probably call these i.e. `latency_ms_p50` so that the unit is explicit.